sdk/iot/hub/samples/src/paho_iot_hub_sas_telemetry_example.c

```diff
@@ -0,0 +1,418 @@
+  az_span iot_hub_sas_expiration = AZ_SPAN_FROM_BUFFER(iot_hub_sas_key_expiration_char);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "IoT Hub Device SAS Key Expiration (Hours)",
+      ENV_IOT_HUB_SAS_KEY_EXPIRATION,
+      "2",
+      false,
+      iot_hub_sas_expiration,
+      &iot_hub_sas_expiration));
+
+  az_span iot_hub_sas_key_expiration_span
+      = az_span_init(az_span_ptr(iot_hub_sas_expiration), az_span_size(iot_hub_sas_expiration));
+  AZ_RETURN_IF_FAILED(az_span_atou32(iot_hub_sas_key_expiration_span, &iot_hub_sas_key_expiration));
+
+  iot_hub_sas_key_span = AZ_SPAN_FROM_BUFFER(iot_hub_sas_key);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "IoT Hub Device SAS Key",
+      ENV_IOT_HUB_SAS_KEY,
+      NULL,
+      true,
+      iot_hub_sas_key_span,
+      &iot_hub_sas_key_span));
+
+  az_span device_id_span = AZ_SPAN_FROM_BUFFER(device_id);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "Device ID", ENV_DEVICE_ID, "", false, device_id_span, &device_id_span));
+
+  az_span iot_hub_hostname_span = AZ_SPAN_FROM_BUFFER(iot_hub_hostname);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "IoT Hub Hostname",
+      ENV_IOT_HUB_HOSTNAME,
+      "",
+      false,
+      iot_hub_hostname_span,
+      &iot_hub_hostname_span));
+
+  // Paho requires that the MQTT endpoint be of the form ssl://<HUB ENDPOINT>:8883
+  AZ_RETURN_IF_FAILED(
+      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+static int get_sas_key()
+{
+  az_result res;
+
+  // Create the POSIX expiration time from input hours
+  uint32_t sas_expiration = get_expiration_time(iot_hub_sas_key_expiration);
+
+  // Decode the base64 encoded SAS key to use for HMAC signing
+  az_span decoded_key_span;
+  if (az_failed(
+          res = sample_base64_decode(
+              iot_hub_sas_key_span, AZ_SPAN_FROM_BUFFER(sas_b64_decoded_key), &decoded_key_span)))
+  {
+    printf("Could not decode the SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // Get the signature which will be signed with the decoded key
+  az_span sas_signature_span;
+  if (az_failed(
+          res = az_iot_hub_client_sas_get_signature(
+              &client,
+              sas_expiration,
+              AZ_SPAN_FROM_BUFFER(sas_signature_buf),
+              &sas_signature_span)))
+  {
+    printf("Could not get the signature for SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // HMAC the signature with the decoded key
+  az_span encoded_span = AZ_SPAN_FROM_BUFFER(sas_signature_hmac_encoded_buf);
```

Comment:
everywhere that says "hmac encoded" should be "hmac sha256 encrypted".

Reply:
plus "encoded_span" is not very representative. 
Sorry Dane, I only have naming comments... :(